pandas/core/base.py

```diff
@@ -954,7 +954,11 @@ def _map_values(self, mapper, na_action=None):
                 # we specify the keys here to handle the
                 # possibility that they are tuples
                 from pandas import Series
-                mapper = Series(mapper)
+                if len(mapper) == 0:
```

Comment:
I think this might be worthwhile to make a function in pandas.core.dtypes.cast.maybe_infer_1d_array_dtype